src/UI/ToggleTheme.jsx

```diff
@@ -34,7 +34,7 @@ const SliderToggle = ({ selected, setSelected }) => {
     <div className="relative flex w-fit items-center rounded-full">
       <button
         className={`${TOGGLE_CLASSES} ${
-          selected === 'light' ? 'text-white' : 'text-slate-300'
+          selected === 'light' ? 'text-foreground' : 'text-slate-300'
```

Comment:
You don't need this type of check, try just 

```jsx
className={`${TOGGLE_CLASSES} ${'text-foreground'}`}
```
and see the magic

So, `foreground` has different values for light/dark theme, and they are applied when the themes switch, that's why you have those CSS variables for colors in `body` for light, and `body.dark` for dark

If you want to set something specific for dark mode, for instance, you can do `dark:bg-red-500` and it will apply red background only when it's dark mode